addons/hr_contract/models/resource.py

```diff
@@ -28,17 +31,35 @@ def transfer_leaves_to(self, other_calendar, resources=None, from_date=None):
             'calendar_id': other_calendar.id,
         })
 
+    @api.depends('contract_ids', 'contract_ids.state')
     def _compute_contracts_count(self):
-        count_data = self.env['hr.contract']._read_group(
-            [('resource_calendar_id', 'in', self.ids)],
-            ['resource_calendar_id'],
-            ['__count'])
-        mapped_counts = {resource_calendar.id: count for resource_calendar, count in count_data}
+        """ Compute total and running contract counts in a single query. """
+        contracts_data = self.env['hr.contract']._read_group(
+            [('resource_calendar_id', 'in', self.ids), ('company_id', '=', self.env.company.id)],
+            ['resource_calendar_id', 'state'],
+            ['__count']
+        )
+        total_contracts_count = defaultdict(int)
+        running_contracts_count = defaultdict(int)
+        for calendar, state, count in contracts_data:
+            if state == 'open':
+                running_contracts_count[calendar.id] = count
+            total_contracts_count[calendar.id] += count
         for calendar in self:
-            calendar.contracts_count = mapped_counts.get(calendar.id, 0)
+            calendar.contracts_count = total_contracts_count.get(calendar.id, 0)
+            calendar.running_contracts_count = running_contracts_count.get(calendar.id, 0)
 
```

Comment:
if your dictionary is a default dict for an integer the default values is 0 so you can remove your get :)
or you can just create a normal dict as you wish